Calculator/CalculatorLibrary/CalculatorLibrary.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics;
+using Newtonsoft.Json;
+
+namespace CalculatorLibrary
+{
+    public class Calculator
+    {
+        JsonWriter writer;
+
+        // Constructor incl opening Json Writer & file for log of operations
+        public Calculator()
+        {
+            StreamWriter logFile = File.CreateText("calculatorlog.json");
+            logFile.AutoFlush = true;
+            writer = new JsonTextWriter(logFile);
+            writer.Formatting = Formatting.Indented;
+            writer.WriteStartObject();
+            writer.WritePropertyName("Operations");
+            writer.WriteStartArray();
+        }
+
+        public double DoOperation(double num1, double num2, string op)
+        {
+            double result = double.NaN;  // Default value is "not-a-number" which we use if an operation such as division could result in an error.
+
+            writer.WriteStartObject();
```

Comment:
Using any writer or reader across methods could result in an evil leaky memory situation. Especially in multi threaded applications, this will BYOTA (Bite you on the arse)